client/trino-client/src/main/java/io/trino/client/uri/PropertyName.java

```diff
@@ -18,66 +18,78 @@
+import static java.util.Objects.requireNonNull;
 import static java.util.function.Function.identity;
 
 public enum PropertyName
 {
-    USER("user"),
-    PASSWORD("password"),
-    SESSION_USER("sessionUser"),
-    ROLES("roles"),
-    SOCKS_PROXY("socksProxy"),
-    HTTP_PROXY("httpProxy"),
-    APPLICATION_NAME_PREFIX("applicationNamePrefix"),
-    DISABLE_COMPRESSION("disableCompression"),
-    ASSUME_LITERAL_NAMES_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS("assumeLiteralNamesInMetadataCallsForNonConformingClients"),
-    ASSUME_LITERAL_UNDERSCORE_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS("assumeLiteralUnderscoreInMetadataCallsForNonConformingClients"),
-    SSL("SSL"),
-    SSL_VERIFICATION("SSLVerification"),
-    SSL_KEY_STORE_PATH("SSLKeyStorePath"),
-    SSL_KEY_STORE_PASSWORD("SSLKeyStorePassword"),
-    SSL_KEY_STORE_TYPE("SSLKeyStoreType"),
-    SSL_TRUST_STORE_PATH("SSLTrustStorePath"),
-    SSL_TRUST_STORE_PASSWORD("SSLTrustStorePassword"),
-    SSL_TRUST_STORE_TYPE("SSLTrustStoreType"),
-    SSL_USE_SYSTEM_TRUST_STORE("SSLUseSystemTrustStore"),
-    KERBEROS_SERVICE_PRINCIPAL_PATTERN("KerberosServicePrincipalPattern"),
-    KERBEROS_REMOTE_SERVICE_NAME("KerberosRemoteServiceName"),
-    KERBEROS_USE_CANONICAL_HOSTNAME("KerberosUseCanonicalHostname"),
-    KERBEROS_PRINCIPAL("KerberosPrincipal"),
-    KERBEROS_CONFIG_PATH("KerberosConfigPath"),
-    KERBEROS_KEYTAB_PATH("KerberosKeytabPath"),
-    KERBEROS_CREDENTIAL_CACHE_PATH("KerberosCredentialCachePath"),
-    KERBEROS_DELEGATION("KerberosDelegation"),
-    KERBEROS_CONSTRAINED_DELEGATION("KerberosConstrainedDelegation"),
-    ACCESS_TOKEN("accessToken"),
-    EXTERNAL_AUTHENTICATION("externalAuthentication"),
-    EXTERNAL_AUTHENTICATION_TIMEOUT("externalAuthenticationTimeout"),
-    EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS("externalAuthenticationRedirectHandlers"),
-    EXTERNAL_AUTHENTICATION_TOKEN_CACHE("externalAuthenticationTokenCache"),
-    EXTRA_CREDENTIALS("extraCredentials"),
-    CLIENT_INFO("clientInfo"),
-    CLIENT_TAGS("clientTags"),
-    TRACE_TOKEN("traceToken"),
-    SESSION_PROPERTIES("sessionProperties"),
-    SOURCE("source"),
-    LEGACY_PREPARED_STATEMENTS("legacyPreparedStatements"),
-    DNS_RESOLVER("dnsResolver"),
-    DNS_RESOLVER_CONTEXT("dnsResolverContext"),
-    HOSTNAME_IN_CERTIFICATE("hostnameInCertificate"),
-    TIMEZONE("timezone"),
+    USER("user", "User name"),
+    PASSWORD("password", "User password"),
+    SESSION_USER("sessionUser", "Name of the user to impersonate"),
+    ROLES("roles", "Access roles that will be used to access catalogs"),
+    SOCKS_PROXY("socksProxy", "SOCKS proxy address"),
+    HTTP_PROXY("httpProxy", "HTTP proxy adress"),
+    APPLICATION_NAME_PREFIX("applicationNamePrefix", "Application name"),
+    DISABLE_COMPRESSION("disableCompression", "Disable compression"),
+    ASSUME_LITERAL_NAMES_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS("assumeLiteralNamesInMetadataCallsForNonConformingClients", "tbd"),
```

Comment:
Maybe `null` instead of `"tbd"`? If we don't have a good human-readable label for this, maybe code that would try to show this to users should throw exceptions.

Reply:
If this is only used in `DriverPropertyInfo`, the docs say it can be null.

Reply:
I'll drop this change entirely.